### PR TITLE
Add autorest and try to fix nyc sourcemap problem

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

File: lib/autorest.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    }
+    result += "autorest";
+  }
+  if (options.osPlatform === "win32" && !result.endsWith(".cmd")) {
+    result += ".cmd";
+  }
+  return result;
+}
+
+export type Switch = "" | boolean;
```
Comment (kpajdzik) [inline lib/autorest.ts]: Why?

Comment (daschult) [reply lib/autorest.ts]: I kept typing `"" | boolean`, so I felt like it deserved a name. It does describe a command-line argument switch that can be described in a readme.md file.

Comment (kpajdzik) [reply lib/autorest.ts]: I get that. Why it can be an empty string? If it's defined like
`csharp` instead of `csharp: true` or something?

Comment (daschult) [reply lib/autorest.ts]: Yup. In some swagger-to-sdk config files you'll see:
```json
{
  "python": ""
}
```
which is the equivalent of specifying `--python` on the command-line.

**Pull request merged**